lib/server/sts.js

```diff
@@ -0,0 +1,1810 @@
+                next(new errors.AccessDeniedError('Invalid LDAP credentials'));
+            } else {
+                next(new errors.InternalError(
+                    'Failed to create session token: ' + addErr.message));
+            }
+            return;
+        }
+
+        log.info({
+            accessKeyId: tempAccessKeyId,
+            expiration: expiration.toISOString(),
+            principal: callerUuid,
+            dn: dn
+        }, 'Successfully created session token in UFDS');
+
+        // Store in Redis for fast lookup (similar to AssumeRole)
+        var accessKeyData = {
+            type: 'accesskey',
+            accessKeyId: tempAccessKeyId,
+            secretAccessKey: tempSecretKey,
+            sessionToken: secureSessionToken,
+            userUuid: callerUuid,
+            expiration: expiration.toISOString(),
+            credentialType: 'temporary',
+            created: Date.now().toString(),
+            // No assumedRole for GetSessionToken
+            // - it's for the original principal
+            principalUuid: callerUuid
+        };
+
+        var redisKey = '/accesskey/' + tempAccessKeyId;
+        var credentialJson = JSON.stringify(accessKeyData);
+
+        req.redis.set(redisKey, credentialJson, function (setErr) {
+            if (setErr) {
+                log.error({
+                    err: setErr,
+                    accessKeyId: tempAccessKeyId
+                }, 'Failed to store GetSessionToken credential in Redis');
+                // Continue anyway - UFDS storage is primary
+            } else {
+                log.info({
+                    accessKeyId: tempAccessKeyId,
+                    expiration: expiration.toISOString()
+                }, 'Successfully stored GetSessionToken credential in Redis');
+            }
+        });
+
+        // Return STS response
+        var response = {
+            GetSessionTokenResponse: {
+                GetSessionTokenResult: {
+                    Credentials: {
+                        AccessKeyId: tempAccessKeyId,
+                        SecretAccessKey: tempSecretKey,
+                        SessionToken: secureSessionToken,
+                        Expiration: expiration.toISOString()
+                    }
+                }
+            }
+        };
+
+        res.send(200, response);
+        next();
+        return;
+        });
+    }); // End pool.acquire callback
+    } // End continueWithCredentialGeneration function
+}
+/*jsl:end*/
+
+/**
+ * @brief GetCallerIdentity STS operation implementation
+ *
+ * Returns details about the IAM entity used to make the request.
+ * This operation does not require any specific permissions and can be
+ * called with any valid AWS credentials.
+ *
+ * @param {Object} req HTTP request object with authentication info
+ * @param {Object} res HTTP response object
+ * @param {function} next Next middleware function
+ *
+ * @returns AWS XML response with caller identity information:
```

Comment:
So STS requires XML... I saw that in a commit message in here, IIRC.  This needs translation because most of Triton operates with JSON objects instead, right?

If there's a commit that answers this, just give me the commit hash or pointer to it and I'll read it myself.